src/Stratis.Bitcoin.IntegrationTests.Common/Extensions/CoreNodeExtensions.cs

```diff
@@ -21,11 +21,17 @@ public static Money GetProofOfWorkRewardForMinedBlocks(this CoreNode node, int n
 
             int startBlock = node.FullNode.Chain.Height - numberOfBlocks + 1;
 
-            return Enumerable.Range(startBlock, numberOfBlocks)
+            int maturity = (int)node.FullNode.Network.Consensus.CoinbaseMaturity;
```

Comment:
The goal of this method is simply to return the reward, whether or not the rewards are spendable or not.
Are you sure this change is necessary? It shouldn't be. If it is then it's the tests using it that need looking into.

Reply:
I spent the last two weeks looking at the tests. This change is the least required to get everything to work.

Reply:
> This change is the least required to get everything to work.

I don't understand how this is related.

Reply:
I am not sure how I am supposed to simultaneously get the tests to pass, and not change anything about them. This method gets used implicitly every time `SharedSteps.MineBlocks` is called, to ensure internal consistency. Its result gets compared to the increase in wallet balance, and any discrepancy causes the test to immediately fail. As the wallet balance is dependent on the underlying bug this PR fixes, it necessitates that this function be changed too. The alternative is to change every test that uses MineBlocks, which results in a lot more changed code & possible structural changes to the tests.

Reply:
This method is used in `GetProofOfWorkRewardForMinedBlocksTest` to check what's the miner reward after mining, but here you've changed the test there to match your change.
And you've added a test called `GetProofOfWorkRewardForMinedBlocksTestLowMaturity` that includes maturity in your calculation but it's not related to the miner reward.
@zeptin @fassadlr